### ISSUE TITLE
Proposal: New modular rule - "Pushing the envelope"

### ISSUE THREAD

**Issue by asmialek** [Wollivan/SquadronLeader#69]
The flight envelope is a graph visualizing the capabilities of the aircraft in terms of min-max speeds and the altitude at which it can safely fly. The envelope itself is of course defined by various parameters and there is absolutely no way of simulating that in a board game. But, there is a term ["Pushing the envelope"](https://en.wikipedia.org/wiki/Flight_envelope#%22Pushing_the_envelope%22), which means that the pilot tries to go over the speed or altitude defined by the flight envelope while risking the integrity of the aircraft (or causing the plane to stall).

![image](https://user-images.githubusercontent.com/14300968/171614523-9f3c17ff-d2ca-43d5-9e56-d1ef1cef7524.png)

To show this, why not add the ability for the daring pilots to be able to "Push the envelope", giving them the ability to either have 1 more Speed for a round or gain 5th altitude token, at the risk of losing some health?

Say:

>## "Pushing the envelope" rule
>
>This rule rewards daring pilots who are willing to test the limits of the aircraft by going faster and higher than it should be possible!
>
>At any point during the move phase a player may declare they are "Pushing the envelope" of their plane and roll one die. On a result of 5-8, they gain the ability to either increase their Speed by 1 for a turn or gain the 5th altitude token (if they already have 4) - but not both. On a result of 1-4, they lose 1 Health instead. The pilot might only "Push the envelope" only once per turn.

I'm not married to the exact wording, but it would give a nice "push your luck" element to the game

**Comment by Wollivan** [comment]
Ah man I really like the flavour of this rule!

I also like the name, again, because it's so flavourful!

I've had a ridiculously crazy week and a bit, so I've got a bunch of minor rules corrections to add, but this goes to the top of the list to be added!

**Comment by Wollivan** [comment]
I totally thought I'd already done this @asmialek !!

Its now in the game as of https://github.com/Wollivan/SquadronLeader/pull/101 <3
